tests/deployment/sagemaker/test_sagemaker.py

```diff
@@ -1,33 +1,419 @@
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": "*",
+                    "Principal": {
+                        "Service": "sagemaker.amazonaws.com"
+                    }
+                }
+            ]
+        }
+        """
+        iam_client = boto3.client('iam', TEST_AWS_REGION)
+        iam_client.create_role(
+            RoleName="moto", AssumeRolePolicyDocument=iam_role_policy
+        )
+        return func(*args, **kwargs)
+
+    return mock_wrapper
+
+
+def mock_sagemaker_deployment_wrapper(func):
+    if version_info.major >= 3:
+        mock_open_param_value = 'builtins.open'
+    else:
+        mock_open_param_value = '__builtin__.open'
+
+    @mock_aws_services_for_sagemaker
+    @patch('subprocess.check_output', autospec=True)
+    @patch('docker.APIClient.build', autospec=True)
+    @patch('docker.APIClient.push', autospec=True)
+    @patch(mock_open_param_value, mock_open(read_data='test'), create=True)
+    @patch('shutil.copytree', autospec=True)
+    @patch('os.chmod', autospec=True)
+    def mock_wrapper(*args, **kwargs):
+        return func(*args, **kwargs)
+
+    return mock_wrapper
+
+
+def create_yatai_service(is_local_repo=True):
+    def mock_get_bento(is_local_repo=True):
+        bento_pb = Bento(
+            name=TEST_DEPLOYMENT_BENTO_NAME, version=TEST_DEPLOYMENT_BENTO_VERSION
+        )
+        # BentoUri.StorageType.LOCAL
+        if is_local_repo:
+            bento_pb.uri.type = 1
+            bento_pb.uri.uri = TEST_DEPLOYMENT_BENTO_LOCAL_URI
+        api = BentoServiceMetadata.BentoServiceApi(name=TEST_BENTO_API_NAME)
+        bento_pb.bento_service_metadata.apis.extend([api])
+        return GetBentoResponse(bento=bento_pb)
+
+    fake_yatai_service = MagicMock()
+    fake_yatai_service.GetBento = lambda pb_request: mock_get_bento(is_local_repo)
+
+    return fake_yatai_service
+
+
+def generate_sagemaker_deployment_pb():
+    test_deployment_pb = Deployment(
+        name=TEST_DEPLOYMENT_NAME, namespace=TEST_DEPLOYMENT_NAMESPACE
+    )
+    test_deployment_pb.spec.bento_name = TEST_DEPLOYMENT_BENTO_NAME
+    test_deployment_pb.spec.bento_version = TEST_DEPLOYMENT_BENTO_VERSION
+    test_deployment_pb.spec.sagemaker_operator_config.api_name = TEST_BENTO_API_NAME
+    test_deployment_pb.spec.sagemaker_operator_config.region = TEST_AWS_REGION
+    test_deployment_pb.spec.sagemaker_operator_config.instance_count = (
+        TEST_DEPLOYMENT_INSTANCE_COUNT
+    )
+    test_deployment_pb.spec.sagemaker_operator_config.instance_type = (
+        TEST_DEPLOYMENT_INSTANCE_TYPE
+    )
+    # DeploymentSpec.DeploymentOperator.AWS_SAGEMAKER
+    test_deployment_pb.spec.operator = 2
+
+    return test_deployment_pb
+
+
+def raise_(ex):
+    raise ex
+
+
+@mock_sagemaker_deployment_wrapper
+def test_sagemaker_apply_fail_not_local_repo(
+    mock_chmod, mock_copytree, mock_docker_push, mock_docker_build, mock_check_output
```

Comment:
since those mocks are referenced here, could we just do a sanity check on the ones that actually expected to get called and omit the ones that are not needed? same for the other a few tests in this file